eva/catalog/catalog_manager.py

```diff
@@ -81,143 +85,265 @@ def _shutdown_catalog(self):
         logger.info("Shutting catalog")
         drop_db()
 
-    def create_metadata(
+    "Table catalog services"
+
+    def insert_table_catalog_entry(
         self,
         name: str,
         file_url: str,
-        column_list: List[DataFrameColumn],
+        column_list: List[ColumnCatalog],
         identifier_column="id",
         table_type=TableType.VIDEO_DATA,
-    ) -> DataFrameMetadata:
-        """Creates metadata object
-
-        Creates a metadata object and column objects and persists them in
-        database. Sets the schema field of the metadata object.
+    ) -> TableCatalog:
+        """A new entry is added to the table catalog and persisted in the database.
+        The schema field is set before the object is returned."
 
         Args:
-            name: name of the dataset/video to which this metdata corresponds
+            name: table name
             file_url: #todo
             column_list: list of columns
             identifier_column (str):  A unique identifier column for each row
             table_type (TableType): type of the table, video, images etc
         Returns:
-            The persisted DataFrameMetadata object with the id field populated.
+            The persisted TableCatalog object with the id field populated.
         """
 
-        metadata = self._dataset_service.create_dataset(
+        table_entry = self._table_catalog_service.insert_entry(
             name,
             file_url,
             identifier_id=identifier_column,
             table_type=table_type,
         )
 
-        # Append row_id to table metadata.
+        # Append row_id to table column list.
         column_list = [
-            DataFrameColumn(IDENTIFIER_COLUMN, ColumnType.INTEGER)
+            ColumnCatalog(IDENTIFIER_COLUMN, ColumnType.INTEGER)
         ] + column_list
 
         for column in column_list:
-            column.metadata_id = metadata.id
-        column_list = self._column_service.create_column(column_list)
+            column.table_id = table_entry.id
+        column_list = self._column_service.insert_entries(column_list)
 
-        metadata.schema = column_list
-        return metadata
+        table_entry.schema = column_list
+        return table_entry
 
-    def create_multimedia_table(self, name: str, format_type: FileFormatType):
-        if format_type is FileFormatType.VIDEO:
-            return self._create_video_table(name)
-        elif format_type is FileFormatType.IMAGE:
-            return self._create_image_table(name)
-        else:
-            raise CatalogError(f"Format Type {format_type} is not supported")
+    def get_table_catalog_entry(
+        self, database_name: str, table_name: str
```

Comment:
Can we drop `database_name`for this function?

Reply:
I gave it a thought. Maybe we need it in the future if we support database-level distinction. How about I make it an optional argument?

Reply:
Sure. Sounds good to me.